src/components/molecules/FormFields/FilesInput/index.tsx

```diff
@@ -20,8 +22,9 @@ export default function FilesInput(props: InputProps): ReactElement {
 
     try {
       setIsLoading(true)
-      const newFileInfo = await getFileInfo(url)
-      newFileInfo && helpers.setValue([newFileInfo])
+      const filesArray: FileMetadata[] = await ocean.provider.fileinfo(url)
```

Comment:
the downside now is that adding a file can't be tested without a wallet, users get this, making them assume their file is not working:

<img width="1305" alt="Screen Shot 2021-01-29 at 15 45 09" src="https://user-images.githubusercontent.com/90316/106289045-120e6500-6249-11eb-8871-36af8c62a9d8.png">

That is always the problem when using the `ocean` object which is why in the `market` we often work around that by hitting some API directly. Haven't checked but maybe we can hit provider API directly? Or is a web3 account really required for this?

If we want to keep using `ocean` here, then we need to adapt the UI for it, like:

- disabling the input (but then we have to tell users why it's disabled)
- adapt the toast error message

Reply:
I believe it better to try to hit the provider api firstly, if we won't be able to use the api without a web3 account i guess it would be better to adapt to toast message depending on the error thrown

Reply:
Well, tried to find new provider endpoints but [they are undocumented](https://github.com/oceanprotocol/provider/issues/76). There's a `/fileinfo` route we could hit directly https://github.com/oceanprotocol/provider/blob/master/ocean_provider/routes.py#L223, so I guess full path would be `/api/v1/services/fileinfo`

Reply:
While I agree with what you said, what is the point of the form without a wallet? If a user is not initiated in the world of web3 and he starts filling the form and at the end realizes that he doesn't have a wallet are we not wasting his time?

Reply:
The use case here would be to allow users exploring, testing their files. In general it's about a UX principle we try to follow in market overall: every functionality we can make work without Web3 and a wallet we should.

Reply:
I agree, in fact we would somehow treat differently the input fields and the user will be able to fill all required fields except file and sample file inputs, so filling name, description is also a waste of time for the user if no wallet is connected.
I will try to hit the provider api directly (initially I was trying to use the provider class from OceanJS )  if we decide to stick with the principle of allowing the users to explore and test their files.

Reply:
Then call `https://provider.mainnet.oceanprotocol.com/api/v1/services/fileinfo` like we directly call other endpoints

Reply:
yup, ocean.js does nothing else so we can mimic that, with the bonus of having our axios cancel token although not super useful in this case https://github.com/oceanprotocol/ocean.js/blob/main/src/provider/Provider.ts#L77-L100

Reply:
but if the wallet is connected we should get the url from the config. Even though it does the same thing we should call the correct provider (based on network ), maybe we are testing something on test network.

Reply:
yes, for the Aquarius replacement methods we always pass the URL for that reason from the component where we get that from `useOcean()`: https://github.com/oceanprotocol/market/blob/main/src/utils/aquarius.ts#L40

Reply:
actually get `config.providerUri` like we get `metadataCacheUri` https://github.com/oceanprotocol/market/blob/ab609b93c0aa41b1dc567bba2163c95b71434828/src/components/pages/Home.tsx#L60 since this works without a wallet.